datamodels/2.x/itop-portal-base/portal/src/Form/ObjectFormManager.php

```diff
@@ -1159,6 +1161,14 @@ public function OnSubmit($aArgs = null)
 				$this->oObject->DBWrite();
 			}
 			catch (Exception $e) {
+				$aContext = [
+					'origin'    => __CLASS__.'::'.__METHOD__,
+					'obj_class' => get_class($this->oObject),
+					'obj_key' => $bIsNew ? 'NULL' : $this->oObject->GetKey(),
```

Comment:
```suggestion
					'obj_key' => $this->oObject->GetKey(),
```

Keep the negative key when object has not been saved yet.